PluginCore/PluginCore/Managers/TraceManager.cs

```diff
@@ -102,28 +102,82 @@ public static void AddAsync(String message, Int32 state)
         /// </summary>
-        /// <param name="group">The id used to determine the group traces</param>
-        /// <param name="title">The title of the group</param>
-        /// <param name="icon">An icon for the group. If this is null, the default icon is used</param>
-        public static void RegisterTraceGroup(string group, string title, Image icon)
+        /// <param name="groupId">The id used to determine the trace group.</param>
+        /// <param name="title">The title to be displayed in the panel.</param>
+        /// <param name="showFilterButtons">Whether to show info, warning and error filter buttons.</param>
+        /// <param name="allowMultiplePanels">Whether to let users lock certain panels to create multiple panels.</param>
+        /// <param name="icon">The icon to be displayed in the panel. If this is <see langword="null"/>, the default icon is used.</param>
+        public static void RegisterTraceGroup(string groupId, string title, bool showFilterButtons = true, bool allowMultiplePanels = false, Image icon = null)
         {
-            traceGroups.Add(group, new TraceGroup(group, title, icon));
+            if (groupId == null) throw new ArgumentNullException(nameof(groupId));
+            traceGroups.Add(groupId, new TraceGroup(groupId, title, showFilterButtons, allowMultiplePanels, icon));
         }
 
-        public static TraceGroup GetTraceGroup(string group)
+        /// <summary>
+        /// Returns the <see cref="TraceGroup"/> with the specified group id, or <see langword="null"/>.
+        /// </summary>
+        /// <param name="groupId">The id of the trace group.</param>
+        public static TraceGroup GetTraceGroup(string groupId)
         {
-            if (group == null) return null;
-
+            if (groupId == null) throw new ArgumentNullException(nameof(groupId));
             TraceGroup value;
-            return traceGroups.TryGetValue(group, out value) ? value : null;
+            return traceGroups.TryGetValue(groupId, out value) ? value : null;
+        }
+
+        /// <summary>
+        /// Creates a string for group data.
+        /// <para/>
+        /// <paramref name="groupId"/>:arg1,arg2,<paramref name="args"/>
+        /// </summary>
+        public static string CreateGroupData(string groupId, params string[] args)
+        {
+            if (args == null || args.Length == 0)
+            {
+                return groupId;
+            }
+            return groupId + ":" + string.Join(",", args);
+        }
+
+        /// <summary>
+        /// Creates a string for group data with a unique id argument at the end. Used for groups with <see cref="TraceGroup.AllowMultiplePanels"/> set to <see langword="true"/>.
+        /// <para/>
+        /// <paramref name="groupId"/>:arg1,arg2,<paramref name="args"/>,uniqueId
+        /// </summary>
+        public static string CreateGroupDataUnique(string groupId, params string[] args)
```

Comment:
I didn't have it at the start, but as I was writing implementations, such as in Find All References, it was quite inconvenient, since i had to manually have a static variable for each implementation, and increment and pass the value.

One reason was to have only one static variable for the unique token.
Second reason was that the callers never really need to access this unique token unless wanted, and the value of the token is meaningless. It means they wouldn't really want to set or store the unique token themselves.